sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala

```diff
@@ -93,7 +93,7 @@ class ContinuousExecution(
   }
 
   private val triggerExecutor = trigger match {
-    case ContinuousTrigger(t) => ProcessingTimeExecutor(ProcessingTime(t), triggerClock)
+    case ContinuousTrigger(t) => ProcessingTimeExecutor(ProcessingTimeTrigger(t), triggerClock)
```

Comment:
Should this use `Trigger.ProcessingTime` instead of this impl class directly or do I misunderstand?

Reply:
Trigger.xxx returns `Trigger`, whereas it requires `ProcessingTimeTrigger` (was requiring `ProcessingTime`). From user-facing API we avoid dealing with actual implementations via only exposing `Trigger` but internally we require actual implementations.

Reply:
OK, so this 'processing time' implementation requires the concrete implementation of the trigger. If so I get it.